File: app/common/domain/requests/agent.py
from typing import Dict, Any, Optional

from pydantic import BaseModel, Field

from app.common.domain.enums.agent_category import AgentCategory
from app.common.domain.requests.base import BasePageRequest


class PageAgentRequest(BasePageRequest):
    ...


class AddAgentRequest(BaseModel):
    name: str
    model: str
    instruction: str
    graph_id: str
    category: AgentCategory = Field(default=AgentCategory.GENERAL)
    structured_output_schema: Dict[str, Any]


class UpdateAgentRequest(BaseModel):
    name: Optional[str] = Field(default=None)
    model: Optional[str] = Field(default=None)
    instruction: Optional[str] = Field(default=None)
    graph_id: Optional[str] = Field(default=None)
    category: Optional[AgentCategory] = Field(default=None)
    structured_output_schema: Optional[Dict[str, Any]] = Field(default=None)

File: web/src/types/agent.ts
export type AgentDetail = {
	name: string;
	model: string;
	instruction: string;
	category: string;
	tools: string[];
	structured_output_schema: {
		type: string;
		properties: {
			[key: string]: {
				type: string;
				description: string;
			}[];
		};
	};
};

export type AgentToolDetail = {
	name: string;
	description: string;
	input_schema: Record<string, any>;
	output_schema: Record<string, any>;
};
